redfish/provider/common.go

```diff
@@ -76,6 +77,11 @@ func RedfishServerSchema() map[string]resourceSchema.Attribute {
 			Optional:    true,
 			Description: "This field indicates whether the SSL/TLS certificate must be verified or not",
 		},
+		"redfish_alias": resourceSchema.StringAttribute{
+			MarkdownDescription: redfishAliasMD,
+			Description:         redfishAliasMD,
+			Optional:            true,
```

Comment:
Should we put validators to check that redfish_alias conflicts with all other fields?

Reply:
@rounak-adhikary , user/password are optional, it's ok to use conflicts , but the endpoint required, we cannot use conflicts for it. So thinking to keep user/password and endpoint  consistency, I not use the conflicts for them here.

Reply:
Ah yes, we cannot conflict with a required field.
But we also cannot have users provide both `redfish_alias` and `endpoint` at the same time as that would be an invalid input.
So can we demote `endpoint` to an optional field, and put a validator saying either the `endpoint` or `redfish` alias is required?

Reply:
sure @rounak-adhikary , we can do this. Will demote endpoint to optional and add validation that  one of them is required. Thanks.

Reply:
hi @rounak-adhikary , the improvement is added. Please help review.
And i run the AT for user resource, all existing at cases are passed.
![image](https://github.com/user-attachments/assets/7f44fd16-9470-43ab-a572-ad7c995f28a0)